public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -69,18 +130,58 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   } = useFormContext<RuleFormValues>();
 
   const { queryPreviewData, runQueries, cancelQueries, isPreviewLoading, clearPreviewData } = useAlertQueryRunner();
+  const isSwitchModeEnabled = config.featureToggles.alertingQueryAndExpressionsStepMode ?? false;
 
   const initialState = {
     queries: getValues('queries'),
   };
 
   const [{ queries }, dispatch] = useReducer(queriesAndExpressionsReducer, initialState);
+
+  // data queries only
+  const dataQueries = useMemo(() => {
+    return queries.filter((query) => !isExpressionQuery(query.model));
+  }, [queries]);
+
+  // expression queries only
+  const expressionQueries = useMemo(() => {
+    return queries.filter((query) => isExpressionQuery(query.model));
+  }, [queries]);
+
   const [type, condition, dataSourceName] = watch(['type', 'condition', 'dataSourceName']);
 
   const isGrafanaAlertingType = isGrafanaAlertingRuleByType(type);
   const isRecordingRuleType = isCloudRecordingRuleByType(type);
   const isCloudAlertRuleType = isCloudAlertingRuleByType(type);
 
+  const [isAdvancedMode, setIsAdvancedMode] = useState(isGrafanaAlertingType && isSwitchModeEnabled ? false : true);
+
+  const expressionQueriesList = useMemo(() => {
+    return queries.reduce((acc: ExpressionQuery[], query) => {
+      return isExpressionQuery(query.model) ? acc.concat(query.model) : acc;
+    }, []);
+  }, [queries]);
+  const [showResetModeModal, setShowResetModal] = useState(false);
+
+  const [simpleCondition, setSimpleCondition] = useState<SimpleCondition>(
```

Comment:
nit: I wonder if we can use `useToggle` here, but a simple `useState` is fine too

Reply:
I'm not using `useToggle` because `simpleCondition` is not a `boolean`

Reply:
sorry, maybe you mean for the `showResetModeModal` 👍

Reply:
Oh I did yes, must have tagged the wrong line of code here :D

Reply:
I will keep the useState in this case if you don't mind 👍